tfx/extensions/google_cloud_kubernetes/pusher/executor.py

```diff
@@ -0,0 +1,185 @@
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Custom executor to push TFX model to Kubernetes."""
+
+from os import path
+from typing import Any, Dict, List, Text
+
+from absl import logging
+from kubernetes.client import rest
+from tfx import types
+from tfx.components.pusher import executor as tfx_pusher_executor
+from tfx.types import artifact_utils
+from tfx.utils import json_utils
+from tfx.utils import path_utils
+from tfx.utils import kube_utils
+
+import yaml
+
+# Tensorflow Serving model path format.
+# https://www.tensorflow.org/tfx/serving/serving_kubernetes
+_TFSERVING_MODEL_VERSION_PATH_FORMAT = (
+    '/models/{model_name}/{model_version}')
+
+# Keys to the items in custom_config passed as a part of exec_properties.
+TF_SERVING_ARGS_KEY = 'tf_serving_args'
+
+# Keys for custom_config.
+_CUSTOM_CONFIG_KEY = 'custom_config'
+
+# Number of serving replicas.
+NUM_REPLICAS_KEY = 'num_replicas'
+
+# Name of serving model.
+MODEL_NAME_KEY = 'model_name'
+
+# Model uri environment variable key.
+_MODEL_URI_ENV_KEY = 'MODEL_URI'
+
+# Model name environment variable key.
+_MODEL_NAME_ENV_KEY = 'MODEL_NAME'
+
+# Model base path environment variable key.
+_MODEL_BASE_PATH_ENV_KEY = 'MODEL_BASE_PATH'
+
+class Executor(tfx_pusher_executor.Executor):
+  """Deploy a model to GKE cluster with Tensorflow Serving."""
+
+  def DeployTFServingService(self) -> None:
+    """Creates the model serving service with TF Serving."""
+    client_api = kube_utils.make_core_v1_api()
+    with open(path.join(path.dirname(__file__),
+                        "yaml", "serving-service.yaml")) as f:
+      svc = yaml.safe_load(f)
+      try:
```

Comment:
This doesn't need to be in the with indented block.

Reply:
Fixed!